src/main/java/org/folio/dcb/service/impl/CirculationRequestServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package org.folio.dcb.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dcb.client.feign.CirculationRequestClient;
+import org.folio.dcb.domain.dto.CirculationRequest;
+import org.folio.dcb.service.CirculationRequestService;
+import org.folio.dcb.utils.DCBConstants;
+import org.folio.dcb.utils.RequestStatus;
+import org.folio.spring.FolioExecutionContext;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+import java.util.UUID;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class CirculationRequestServiceImpl implements CirculationRequestService {
+
+  private final CirculationRequestClient circulationStorageClient;
+  private final FolioExecutionContext folioExecutionContext;
+
+  @Override
+  public CirculationRequest fetchRequestById(String requestId) {
```

Comment:
If you agree with the comment provided on the circulationRequestService, we can make this method private